src/gerrit_argo_gateway/__init__.py

```diff
@@ -14,7 +14,19 @@
 
 
 class GerritGateway:
-    _RECHECK_RE = re.compile(r"\s+recheck\s*$", re.I + re.M)
+    # Regular expression for: "Support recheck to request re-running a test."
+    # We only allow blanks or optionally a `sap-openstack-ci` before the `recheck` keyword
+    # The text needs to either end after the `recheck` or at least have a whitespace
+    # to allow for comments by the person triggering the recheck
+    # Positive examples: (Ignoring the header "Patch Set \d:\n\n")
+    #   recheck
+    #   recheck - not sure why it failed
+    #   sap-openstack-ci recheck
+    # Negative examples:
+    #   someotherci recheck
+    #   rechecking
+    _RECHECK_RE = re.compile(r"^.*\n(\s*:?sap-openstack-ci\s+)?\s*recheck(:?[\s]|$)", re.I)
```

Comment:
What's the first `:?` for? Did it move away from the `(` when you added the `\s*`?

Reply:
True, I fixed it.